template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ConfigUtils.java

```diff
@@ -85,4 +89,19 @@ public static String getDbTypePostgres(RdsView rdsView, String serviceType) {
         }
         return DatabaseVendor.POSTGRES.name();
     }
+
+    public static String getDefaultFileSystemUri(CmTemplateProcessor templateProcessor, TemplatePreparationObject templatePreparationObject) {
+        if (HdfsRoleConfigProvider.isNamenodeHA(templatePreparationObject)) {
+            String nameService = templateProcessor.getRoleConfig(HdfsRoles.HDFS, HdfsRoles.NAMENODE, "dfs_federation_namenode_nameservice")
+                    .map(ApiClusterTemplateConfig::getValue)
+                    .orElse(HdfsRoleConfigProvider.DEFAULT_NAME_SERVICE);
+            return "hdfs://" + nameService;
+        }
+
+        Set<String> namenodeHosts = HdfsRoleConfigProvider.nameNodeFQDNs(templatePreparationObject);
+        if (namenodeHosts.size() == 1) {
+            return "hdfs://" + namenodeHosts.iterator().next();
+        }
+        return "";
+    }
```

Comment:
I dont think this is required since by default all the services are configured to use hdfs by default in CM and invoking HdfsRoleConfigProvider from ConfigUtil is not recommended. The order of invocation of these configproviders is also not guaranteed. Please test the default value when cluster is launched without cloud storage before overriding these to non-cloud storage values.

Reply:
I thought except Ranger all other services write it to the local filesystem. I followed the same code as of Ranger. 
https://github.com/hortonworks/cloudbreak/blob/master/template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ranger/RangerCloudStorageServiceConfigProvider.java#L45-L58
Basically we need the FileSystemURI as "hdfs:///" or "s3a://BUCKET". 
As the code was repeating across all the services, I moved it to ConfigUtil.

Reply:
No, for Ranger this field is not initialized by CM, hence ranger has explicit initialization. The values in cloudbreak configproviders are only for overriding default CM values, what is the default value for this service as initialized by CM ? Default values which map to hdfs should be initialized in CM, if it is not being done can be raised as CM Jira.